File: PSP/PrimerApiRest2/PrimerApi/src/main/java/domain/modelo/pokemon/Pokemon.java
package domain.modelo.pokemon;

import lombok.Getter;

import java.util.List;


@Getter
public class Pokemon {
    private int id;
    private String name;
    private int height;
    private int weight;
    private List<StatsItem> stats;
    private Sprites sprites;

    @Override
    public String toString() {
        return "Pokemon{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", height=" + height +
                ", weight=" + weight +
                ", stats=" + stats +
                ", sprites=" + sprites +
                '}';
    }
}

File: movilesPracticas/app/src/main/java/com/example/ejerciciodatajsonlocal/domain/model/Pokemon.kt
package com.example.ejerciciodatajsonlocal.domain.model

import android.os.Parcelable
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import kotlinx.parcelize.Parcelize

@JsonClass(generateAdapter = true)
@Parcelize
data class Pokemon(
    @Json(name = "id")
    val id: Int,
    @Json(name = "name")
    val nombre: String,
    @Json(name = "base_experience")
    val experienciaBase: Int,
    @Json(name = "height")
    val altura: Int,
    @Json(name = "weight")
    val peso: Int,
    @Json(name = "sprites")
    val imagen: String,
    @Json(name = "types")
    val tipoPokemon: String,
) : Parcelable
